src/osiris_util.erl

```diff
@@ -292,4 +295,20 @@ cache_reader_context(Pid, Dir, Name, Shared, Ref, ReadersCounterFun)
                       {Pid, Dir, Name, Shared, Ref, ReadersCounterFun}),
     ok.
 
+is_dir(Dir) ->
+    case prim_file:read_file_info(Dir) of
+        {ok, #file_info{type=directory}} ->
+            true;
+        _ ->
+            false
+    end.
 
+is_file(File) ->
+    case prim_file:read_file_info(File) of
+        {ok, #file_info{type = directory}} ->
+            true;
```

Comment:
false? Copy-paste from `is_dir()`

Reply:
Possibly. I cannot immediately think of a reason to treat directories as "files" here.

Note that this function is only used in one test in [osiris_log_SUITE](https://github.com/rabbitmq/osiris/blob/3376967204014c6cab7d5248056324c53b5bac98/test/osiris_log_SUITE.erl#L1011) and it is used with a "regular" file.

Reply:
This function can be removed. Apart from being wrong it was copied from Ra and never really used.